# Download Dataset

Let's begin by downloading and extracting the dataset.

To download the full dataset, use this URL: https://archive.org/details/egyptian_fruit_bat
To download the 10k flavor of the dataset, use this URL: https://archive.org/details/egyptian_fruit_bats_10k

All other steps remain the same.

In [4]:
!mkdir data
!cd data && curl -L -o egyptian_fruit_bats.zip https://archive.org/download/egyptian_fruit_bats_10k/egyptian_fruit_bats.zip
!cd data && unzip -q egyptian_fruit_bats.zip

mkdir: data: File exists
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 5003M  100 5003M    0     0  13.7M      0  0:06:04  0:06:04 --:--:-- 16.3M0 10.4M     0  12.3M      0  0:06:43  0:02:56  0:03:47 19.4M   46 2320M    0     0  12.6M      0  0:06:34  0:03:03  0:03:31 20.0M     0  0:06:30  0:03:06  0:03:24 21.3M0  13.0M      0  0:06:23  0:03:16  0:03:07 15.8M0:02:29 17.1M 0     0  13.4M      0  0:06:13  0:03:45  0:02:28 16.7M      0  0:06:10  0:03:49  0:02:21 18.5M 0:04:02  0:02:00 19.6M     0  0:05:58  0:04:09  0:01:49 18.3M     0  14.3M      0  0:05:47  0:04:44  0:01:03 16.8M:00:43 9221k 0  13.9M      0  0:05:58  0:05:27  0:00:31 7974k  0:06:04  0:06:01  0:00:03 16.1M


And we are all set!

In [10]:
# should contain audio files and annotations CSV
!ls data

annotations.csv         audio                   egyptian_fruit_bats.zip


`annotations.csv` contains the labels, while the audio files are stored under `data/audio`

In [11]:
# number of audio files in data/audio
!ls -lht data/audio | wc -l

   10001


A total of 10001 bat vocalizations, all recorded at 250_000 Hz.

In [14]:
!ls data/audio | sort -n | head

419.wav
2757.wav
2831.wav
2832.wav
2890.wav
2891.wav
2913.wav
3168.wav
3174.wav
3193.wav
sort: Broken pipe


In [ ]:
# source .venv/bin/activate
# pip install pandas
import pandas as pd

# Note from Mahika: want to use annotations_filenames.csv instead of annotations.csv
anno = pd.read_csv('data/annotations.csv')

In [ ]:
# Emitter and File Name
anno.head()

Emitter  File Name
0      216  69809.wav
1      215  71889.wav
2      216  46690.wav
3      230  85411.wav
4      215  45609.wav

Each of the audio files has been assigned six labels. The files are associated with labels via the file name.